src/lightning/fabric/loggers/csv_logs.py

```diff
@@ -59,6 +63,7 @@ def __init__(
         name: str = "lightning_logs",
         version: Optional[Union[int, str]] = None,
         prefix: str = "",
+        sub_dir: Optional[_PATH] = None,
```

Comment:
Could you open a separate issue for this and only keep the fix for the auto-increment in this PR? Then we can merge the fix faster. Thanks

Reply:
This PR is actually to fix these bug #17886.  This works well in locally. It fails  to pass in PR test cases

Reply:
To fix the auto-increment problem, it shouldn't be necessary to introduce new API like `sub_dir`. I'm simply asking if we could have separate PRs for these two things.

Reply:
It would be easier for me to help with this PR if it was focused on the fix.

Reply:
Sorry for delayed reply.

Actually Problem this issue arise with hidden test checks in GitHub. Better I will close this pull request so you can work on it..